plugins/services/src/js/components/modals/ServiceRootGroupModal/index.tsx

```diff
@@ -0,0 +1,431 @@
+
+import GroupModalHeader from "./Header";
+import ErrorsPanel from "./ErrorsPanel";
+import {
+  emptyGroupFormData,
+  getPathFromGroupId,
+  validateGroupFormData
+} from "./utils";
+
+const dl = container.get<DataLayer>(DataLayerType);
+const i18n = container.get<I18n>(TYPES.I18n);
+
+const groupCreateMutation = gql`
+  mutation {
+    createGroup(data: $data)
+  }
+`;
+
+function getSaveAction(data: GroupFormData) {
+  return dl.query(groupCreateMutation, {
+    data
+  });
+}
+
+interface ServiceRootGroupModalState {
+  isOpen: boolean;
+  isPending: boolean;
+  expandAdvancedSettings: boolean;
+  data: GroupFormData;
+  errors: GroupFormErrors;
+}
+
+interface ServiceRootGroupModalProps {
+  isEdit: boolean;
+}
+
+const METHODS_TO_BIND: string[] = [
+  "getAdvancedSettings",
+  "getModalContent",
+  "handleAdvancedSectionClick",
+  "handleClose",
+  "handleFormChange",
+  "handleSave"
+];
+
+class ServiceRootGroupModal extends React.Component<
+  ServiceRootGroupModalProps,
+  ServiceRootGroupModalState
+> {
+  static contextTypes = {
+    router: routerShape
+  };
+  static defaultProps = {
+    isEdit: false
+  };
+
+  constructor() {
+    // @ts-ignore
+    super(...arguments);
+
+    this.state = this.getInitialState();
+
+    METHODS_TO_BIND.forEach(method => {
+      // @ts-ignore
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getInitialState(
+    _props: ServiceRootGroupModalProps = this.props
+  ): ServiceRootGroupModalState {
+    return {
+      isOpen: true,
+      isPending: false,
+      expandAdvancedSettings: false,
+      data: emptyGroupFormData(),
+      errors: {}
+    };
+  }
+
+  handleClose() {
+    // Start the animation of the modal by setting isOpen to false
+    this.setState(
+      { isOpen: false, isPending: false, data: emptyGroupFormData() },
+      () => {
+        // Once state is set, start a timer for the length of the animation and
+        // navigate away once the animation is over.
+        setTimeout(this.context.router.goBack, 300);
+      }
+    );
+  }
+
+  handleSave() {
+    if (this.state.isPending) {
+      return;
+    }
+    const errors = validateGroupFormData(this.state.data, this.props.isEdit);
+    if (errors) {
+      this.setState({ errors });
+      return;
+    }
+    this.setState({ isPending: true });
+    getSaveAction(this.state.data)
+      .pipe(take(1))
+      .subscribe({
+        next: () => this.handleClose(),
+        error: e => {
+          //TODO: If error creating quota switch to Edit Mode for group, must
+          // Be done after edit mode supported.
+          switch (e.message) {
+            case "Conflict - Group":
+              this.setState({
+                errors: {
+                  id: <Trans>Name already exists. Try a different name.</Trans>,
+                  form: [
+                    <Trans key="groupIdConflict">
+                      A group with the same name already exists. Try a different
```

Comment:
If I try a group with a duplicate name, I don't get this message, but rather:
![Screenshot from 2019-07-29 12-51-11](https://user-images.githubusercontent.com/40791275/62039037-8ead2700-b1ff-11e9-94ca-360b285a6705.png)

Reply:
I updated from `Conflict` to `Conflict - Groups` I thought I got the switch statement in the component too. If not please fix.